app/Theme/FixNonExistentAuthors.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace GovUKBlogs\Theme;
+
+class FixNonExistentAuthors implements \Dxw\Iguana\Registerable
+{
+	public function register()
+	{
+		add_action('gds_byline', [$this, 'replaceAbsentAuthor']);
+	}
+
+	public function replaceAbsentAuthor()
+	{
+		global $post;
+		$post_author_id = get_post_field('post_author', $post->ID);
+		if ($post_author_id > 1) {
+			$post_author = get_user_by('id', $post_author_id);
+			if (!($post_author)) {
+				error_log("author of post {$post->ID} is deleted user $post_author_id", 0);
+				add_filter('wp_insert_post_data', [$this, 'setArchiveAuthor'], 99, 1);
+				wp_update_post($post);
+			}
+		}
+	}
+
+	public function setArchiveAuthor($postData)
+	{
+		$fix_types = ["page", "post", "attachment"];
+
+		if (!in_array($postData['post_type'], $fix_types)) {
+			return $postData;
+		}
+
+		$archive_user_option = get_option('archive_author');
```

Comment:
Question: do we intend to set a default value for this? If not are we intending to go through the sites automatically or manually to set this value?

Reply:
Good point. I'd been thinking that the `wp_options` table applies across the entire network, but of course it doesn't, it's per site. I'll modify this so it takes this value from a network option instead, so we'll have to manually set it, but just the once.

Reply:
Ah right, yes, that was Lee's assumption too!